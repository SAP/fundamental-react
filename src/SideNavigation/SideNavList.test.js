import { mount } from 'enzyme';
import React from 'react';
import { SideNav } from '../';

describe('<SideNavList />', () => {

    describe('SideNavList', () => {
        test('should allow customization of header level', () => {
            const element = mount(<SideNav.List title='test' />);

            expect(
                element.find('.fd-nested-list__group-header').type()
            ).toBe('li');
        });
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the SideNavList component', () => {
            const element = mount(<SideNav.List data-sample='Sample'>
                <SideNav.ListItem
                    id='item-1'
                    name='Link Item 1'
                    url='#' />
                <SideNav.ListItem
                    id='item-2'
                    name='Link Item 2'
                    url='#' />
                <SideNav.ListItem
                    id='item-3'
                    name='Link Item 3'
                    url='#' />
            </SideNav.List>);

            expect(
                element.find('.fd-nested-list').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });

        test('should allow props to be spread to the SideNavList\'s heading element', () => {
            const element = mount(<SideNav.List title='test' titleProps={{ 'data-sample': 'Sample' }} />);

            expect(
                element.find('.fd-nested-list__group-header').getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
