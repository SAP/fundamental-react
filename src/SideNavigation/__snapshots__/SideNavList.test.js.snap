// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNavList /> create side navigation 1`] = `
<div
  className="fd-side-nav__main-navigation"
>
  <ul
    className="fd-nested-list--text-only fd-nested-list level-1"
  >
    <li
      className="fd-nested-list__item"
    >
      <a
        className="fd-nested-list__link"
        href="#"
        onClick={[Function]}
      >
        <span
          className="fd-nested-list__title"
        >
          Item 1
        </span>
      </a>
    </li>
    <li
      className="fd-nested-list__item"
    >
      <a
        className="fd-nested-list__link"
        href="#"
        onClick={[Function]}
      >
        <span
          className="fd-nested-list__title"
        >
          Item 2
        </span>
      </a>
    </li>
    <li
      className="fd-nested-list__item"
    >
      <a
        className="fd-nested-list__link"
        href="#"
        onClick={[Function]}
      >
        <span
          className="fd-nested-list__title"
        >
          Item 3
        </span>
      </a>
    </li>
    <li
      className="fd-nested-list__item"
    >
      <a
        className="fd-nested-list__link"
        href="#"
        onClick={[Function]}
      >
        <span
          className="fd-nested-list__title"
        >
          Item 4
        </span>
      </a>
    </li>
  </ul>
</div>
`;
